boards/nxp/mimxrt1170_evk/mimxrt1170_evk_mimxrt1176_cm7_B.overlay

```diff
@@ -79,3 +80,25 @@
 		status = "okay";
 	};
 };
+
+m2_hci_uart: &lpuart2 {
+	pinctrl-0 = <&pinmux_lpuart2_flowcontrol>;
+	pinctrl-1 = <&pinmux_lpuart2_sleep>;
+	pinctrl-names = "default", "sleep";
+
+	m2_bt_module {
+		compatible = "nxp,bt-hci-uart";
+		sdio-reset-gpios = <&gpio9 15 GPIO_ACTIVE_HIGH>;
+		w-disable-gpios = <&gpio9 30 GPIO_ACTIVE_HIGH>;
+		hci-operation-speed = <115200>;
+		hw-flow-control;
+		fw-download-primary-speed = <115200>;
+		fw-download-secondary-speed = <3000000>;
+		fw-download-secondary-flowcontrol;
+	};
+};
```

Comment:
Just a note here, as part of #72323 we'll probably need to do the same as for the `cyw43xxx.c` driver (see https://github.com/zephyrproject-rtos/zephyr/pull/72323#issuecomment-2147417411), i.e. place a `zephyr,bt-hci-uart` node between the UART node and the `nxp,bt-hci-uart` node (i.e. the UART node would become the grandparent rather than the parent of this driver, which requires some small code changes).